layers/+lang/zig/funcs.el

```diff
@@ -0,0 +1,36 @@
+;;; funcs.el --- zig Layer functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Riccardo Binetti <rbino@gmx.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defun spacemacs//zig-backend ()
+  "Returns selected backend."
+  (if zig-backend
+      zig-backend
+    (cond
+     ((configuration-layer/layer-used-p 'lsp) 'zls)
+     (t nil))))
+
+(defun spacemacs//zig-setup-backend ()
+  "Conditionally setup zig backend."
+  (pcase (spacemacs//zig-backend)
+    (`zls (spacemacs//zig-setup-zls))))
+
+;; lsp
+
+;; zls
+
+(defun spacemacs//zig-setup-zls ()
+  "Setup LSP zls."
+  (if (configuration-layer/layer-used-p 'lsp)
+      (progn
+        (when zls-path
+          (add-to-list 'exec-path zls-path))
```

Comment:
This looks like a global change to the exec-path, don't you have a variable in lsp mode you can set for this? Changing this globally for the need of one layer does not look like an ideal solution to me.

Reply:
`lsp-mode` just [added said variable](https://github.com/emacs-lsp/lsp-mode/pull/2724), so I removed this